query/src/main/java/org/infinispan/search/mapper/search/loading/context/impl/InfinispanSelectionLoadingStrategy.java

```diff
@@ -0,0 +1,17 @@
+package org.infinispan.search.mapper.search.loading.context.impl;
+
+import org.hibernate.search.mapper.pojo.loading.spi.PojoLoadingTypeContext;
+import org.hibernate.search.mapper.pojo.loading.spi.PojoSelectionEntityLoader;
+import org.hibernate.search.mapper.pojo.loading.spi.PojoSelectionLoadingContext;
+import org.hibernate.search.mapper.pojo.loading.spi.PojoSelectionLoadingStrategy;
+
+import java.util.Set;
+
+public class InfinispanSelectionLoadingStrategy<E> implements PojoSelectionLoadingStrategy<E> {
```

Comment:
Be aware that not implementing equals/hashcode here could have performance implications.
That's why I suggested making this a singleton.